cmake/compiler/arcmwdt/generic.cmake

```
@@ -2,21 +2,9 @@
 
 # Configures CMake for using ccac
 
-# MWDT compiler (CCAC) can't be used for preprocessing the DTS sources as it has
-# weird restrictions about file extensions. Synopsys Jira issue: P10019563-38578
-# Let's temporarily use GNU compiler instead.
-find_program(CMAKE_DTS_PREPROCESSOR arc-elf32-gcc)
-if (NOT CMAKE_DTS_PREPROCESSOR)
-  find_program(CMAKE_DTS_PREPROCESSOR arc-linux-gcc)
-endif()
-
-if (NOT CMAKE_DTS_PREPROCESSOR)
-  find_program(CMAKE_DTS_PREPROCESSOR gcc)
-endif()
-
-if(NOT CMAKE_DTS_PREPROCESSOR)
-  message(FATAL_ERROR "Zephyr was unable to find any GNU compiler (ARC or host one) for DTS preprocessing")
-endif()
+set(CMAKE_DTS_PREPROCESSOR ${ARCMWDT_HELPER_DTS_PREPROCESSOR})
```

similar to `CMAKE_OBJCOPY`.

With proposed changes, same principle can be applied:
```suggestion
find_program(CMAKE_DTS_PREPROCESSOR ${ZEPHYR_CROSS_COMPILE}gcc PATHS ${ZEPHYR_SDK_INSTALL_DIR} NO_DEFAULT_PATH)
```